nautobot/ui/src/components/Retrieve/RenderHeader.js

```diff
@@ -59,24 +58,50 @@ export default function RenderHeader({ data }) {
                 )}
             </Heading>
             <ButtonGroup alignItems="center">
-                <UIButton
-                    size="sm"
-                    variant="primaryAction"
-                    leftIcon={<MeatballsIcon />}
-                    onClick={onOpen}
-                >
-                    Actions
-                </UIButton>
-
-                <Modal isOpen={isOpen} onClose={onClose}>
-                    <ModalOverlay />
-
-                    <ModalContent>
-                        <ModalCloseButton />
-
-                        <ModalBody>To be implemented!</ModalBody>
-                    </ModalContent>
-                </Modal>
+                <Menu>
+                    <MenuButton
+                        as={UIButton}
+                        size="sm"
+                        variant="primaryAction"
+                        leftIcon={<MeatballsIcon />}
+                    >
+                        Actions
+                    </MenuButton>
+                    <MenuList>
+                        <MenuItem
+                            to={`${window.location.pathname}edit/`}
+                            icon={<EditIcon />}
+                            onClick={(e) => {
+                                e.preventDefault();
+                                // Because there is currently no support for Edit view in the new UI for production,
+                                // the code below checks if the app is running in production and redirects the user to
+                                // the Edit page; after the page is reloaded, nautobot takes care of rendering the legacy UI.
+                                if (process.env.NODE_ENV === "production") {
```

Comment:
Not sure I quite understand why this is specific to the production env - shouldn't we do this in dev as well?

Reply:
This is more of a copy pasta from https://github.com/nautobot/nautobot/blob/536d75dd802044d5857578748fde58173ca33a26/nautobot/ui/src/components/ObjectTable/ObjectListTable.js#L181

Reply:
Let's augment these with TODO comments for "Hey rip this out when we've addressed this", please.